src/peer_manager.rs

```diff
@@ -276,19 +282,20 @@ impl PeerManager {
 
     /// Clears the routing table and resets this node's public ID.
     pub fn reset_routing_table(&mut self, our_public_id: PublicId, groups: &[Group]) {
+        self.unknown_peers.clear();
+        self.expected_peers.clear();
         let min_group_size = self.routing_table.min_group_size();
         self.our_public_id = our_public_id;
-        let groups_as_names = groups.into_iter()
-            .map(|&(ref prefix, ref members)| {
-                (*prefix, members.into_iter().map(|pub_id| *pub_id.name()).collect_vec())
-            })
-            .collect_vec();
+        self.expected_peers.extend(groups.iter()
```

Comment:
Avoid copying what, the names? But we can't extract those from the `PublicId`s they are contained in without copying, anyway.